packages/block-library/src/common.scss

```diff
@@ -162,6 +162,15 @@ html :where(img[class*="wp-image-"]) {
 	margin: 0 0 1em 0;
 }
 
+/**
+ * Applies box-sizing: border-box; to elements with a class containing 'wp-block-'
+ * and a style attribute containing 'border'.
+ * This ensures consistent box-sizing behavior and makes layout calculations more predictable.
+ */
+[class*="wp-block-"][style*="border"] {
```

Comment:
One catch with this selector is that it would only apply to blocks with border styles applied on the block instance. If a border was applied to a block type via global styles it wouldn't have the correct box-sizing.

This also only covers the application of borders where box-sizing has historically been set for blocks that adopt padding support as well.

Reply:
Thank you for your valuable feedback.I will work on this issue.